nixos/modules/programs/vim.nix

```diff
@@ -17,10 +17,12 @@ in
     package = lib.mkPackageOption pkgs "vim" { example = [ "vim-full" ]; };
   };
 
-  # TODO: convert it into assert after 24.11 release
   config = lib.mkIf (cfg.enable || cfg.defaultEditor) {
-    warnings = lib.mkIf (cfg.defaultEditor && !cfg.enable) [
-      "programs.vim.defaultEditor will only work if programs.vim.enable is enabled, which will be enforced after the 24.11 release"
+    assertions = [
+      {
+        assertion = cfg.defaultEditor && cfg.enable;
```

Comment:
would this complain if it's enabled but not the default editor?

Reply:
Sorry about these mistakes. Should be correct now.